File: 0257-binary-tree-paths/0257-binary-tree-paths.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public void helper(TreeNode root, String s , List<String>ans){
        //base condition
        if(root==null) return;
        String currVal = String.valueOf(root.val);
        if(root.left==null && root.right==null){
            ans.add(s+currVal);
            return;
        }

        helper(root.left, s+currVal+"->",ans);
        helper(root.right,s+currVal+"->",ans);
        return;
    }
    public List<String> binaryTreePaths(TreeNode root) {
        List<String>ans = new ArrayList<>();
        helper(root,"",ans);
        return ans;
    }
}

File: 0257-binary-tree-paths/0257-binary-tree-paths.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void helper(TreeNode* root, string s, vector<string>& ans){
        //base condition
        if(root==NULL) return; //it is not leaf node
        string currVal = to_string(root->val);
        if(root->left==NULL && root->right==NULL){ //main base condition
            //ans me string ko push krna he 
            ans.push_back(s+currVal);
            return;
        }
        
        helper(root->left, s+currVal+"->", ans);
        helper(root->right, s+currVal+"->", ans);
    }
    vector<string> binaryTreePaths(TreeNode* root) {
        vector<string>ans;
        helper(root,"",ans);
        return ans;
    }
};